pytests/func/test_django.py

```diff
@@ -0,0 +1,190 @@
+# Copyright (c) Microsoft Corporation. All rights reserved.
+# Licensed under the MIT License. See LICENSE in the project root
+# for license information.
+
+from __future__ import print_function, with_statement, absolute_import
+
+import os.path
+
+from ..helpers.pattern import ANY
+from ..helpers.timeline import Event
+from .testfiles.testroots import get_test_root
+from ..helpers.webhelper import get_url_from_str, get_web_content
+
+DJANGO1_ROOT = get_test_root('django1')
+DJANGO1_MANAGE = os.path.join(DJANGO1_ROOT, 'app.py')
+DJANGO1_TEMPLATE = os.path.join(DJANGO1_ROOT, 'templates', 'hello.html')
+
+def _wait_for_django_link(debug_session):
+    o = None
+    while True:
+        if o:
+            o = debug_session.wait_for_next(o >> Event('output'))
```

Comment:
You shouldn't need to do `o >> ...` here if you call `session.proceed()` after `wait_for_next` returns. That will specify the cut-off point for the next `wait_for_next` call, below which it won't even look for events.

(There was a bug about this not working right, but it should be fixed in master.)